pkg/grpc/proxy/handler.go

```diff
@@ -119,7 +115,12 @@ func (s *handler) handler(srv interface{}, serverStream grpc.ServerStream) error
 	clientStreamOpts := []grpc.CallOption{
 		grpc.CallContentSubtype((&codec.Proxy{}).Name()),
 	}
+	headersSent := &atomic.Bool{}
```

Comment:
Reviewers: perf improvement here: use a local variable rather than keeping in a sync.Map for better memory usage and perf.